src/k2/module/k23si/Module.cpp

```diff
@@ -526,6 +921,22 @@ K23SIPartitionModule::_createWI(dto::K23SIWriteRequest&& request, std::deque<dto
     return _persistence.makeCall(versions.front(), deadline);
 }
 
+seastar::future<>
+K23SIPartitionModule::_createWI(dto::K23SIPartialUpdateRequest&& request, std::deque<dto::DataRecord>& versions, FastDeadline deadline) {
+    K2DEBUG("Partition: " << _partition << ", PartialUpdate Request creating WI: " << request);
+    dto::DataRecord rec;
+    rec.key = std::move(request.key);
+    // we need to copy this data into a new memory block so that we don't hold onto and fragment the transport memory
+    rec.value = request.value.copy();
```

Comment:
You should explicitly set rec.isTombstone = false here

Reply:
yes,` rec.isTombstone = false` is added.